neurogym/envs/annubes.py

```diff
@@ -136,7 +142,20 @@ def _new_trial(self) -> dict:
         stim_type = None
         stim_value = None
         if not catch:
-            stim_type = self._rng.choice(list(self.session.keys()), p=list(self.session.values()))
+            if len(self.session) == 1:
+                # Single modality task
+                stim_type = next(iter(self.session))
+            elif self.max_sequential is not None and self.sequential_count >= self.max_sequential:
+                # Force a different modality
+                available_modalities = [mod for mod in self.session if mod != self.last_modality]
+                stim_type = self._rng.choice(available_modalities)
+            else:
+                stim_type = self._rng.choice(list(self.session.keys()), p=list(self.session.values()))
+                # Update sequential count
+
+            self.sequential_count = 1 if stim_type != self.last_modality else self.sequential_count + 1
+            self.last_modality = stim_type
+
```

Comment:
As coded, the `max_sequential` will not be applied to catches, meaning there could be many catch trials in a row. I don't think that that is the intent for this setting.

In fact, even for the case of a single non-catch modality, you would still want this setting to apply to not have too many of either the catch or the non-catch situation.

Reply:
> As coded, the `max_sequential` will not be applied to catches, meaning there could be many catch trials in a row. I don't think that that is the intent for this setting.

The limitation is meant for only the modalities, not on the catches. In case of a single non-catch modality, max_sequential does not apply. This was also how it was originally designed in the old annubes (it may have changed when you picked up the implementation).

Reply:
Indeed, at least in my implementation in Annubes, it did apply also to catches. What is the argument to not apply the limitation to catch trials? When thinking about it from the experimentalists point of view, I believe that they would apply the same limitations to a sequence of catch trials as to a non-catch trials.
I think the confusion arises because the order in which we implement things: first we pick a modality, and next we pick whether or not it converts to a catch. In the lab, a catch would be just one of the options (in addition to any of the modalities) for what the next trial could be.

Reply:
I'm pretty sure catch was not seend as a modality in this regard, but we can double check

Reply:
Added as a point to the next meeting agenda